// ngx-siema/__ivy_ngcc__/siema.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxSiemaComponent, NgxSiemaSlideComponent } from './siema.component';
import { NgxSiemaService } from './siema.service';
import * as ɵngcc0 from '@angular/core';
var NgxSiemaModule = (function () {
    function NgxSiemaModule() {
    }
    NgxSiemaModule.forRoot = function () {
        return {
            ngModule: NgxSiemaModule,
            providers: [
                NgxSiemaService,
            ],
        };
    };
NgxSiemaModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxSiemaModule });
NgxSiemaModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxSiemaModule_Factory(t) { return new (t || NgxSiemaModule)(); }, imports: [[
            CommonModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxSiemaModule, { declarations: function () { return [NgxSiemaComponent,
        NgxSiemaSlideComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxSiemaComponent,
        NgxSiemaSlideComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxSiemaModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                ],
                declarations: [
                    NgxSiemaComponent,
                    NgxSiemaSlideComponent,
                ],
                exports: [
                    NgxSiemaComponent,
                    NgxSiemaSlideComponent,
                ]
            }]
    }], function () { return []; }, null); })();
    return NgxSiemaModule;
}());
export { NgxSiemaModule };
NgxSiemaModule.ctorParameters = function () { return []; };
